src/ip_info.rs

```diff
@@ -9,16 +10,25 @@ use reqwest::{
 pub struct IpInfoIoFetch;
 
 impl LocationProvider for IpInfoIoFetch {
-  fn location(&self) -> Result<Location, Error> {
+  fn location(&self, config: &Option<Configuration>) -> Result<Location, Error> {
     let mut json_headers = HeaderMap::new();
     json_headers.insert(CONTENT_TYPE, HeaderValue::from_static("application/json"));
     let client = Client::builder()
       .default_headers(json_headers)
       .gzip(true)
       .use_rustls_tls()
       .build()?;
+
+    let mut url = "https://ipinfo.io/json/".to_string();
+    match config {
+      Some(config) => match config.ip_info_key.borrow() {
+        Some(ip_info_key) => url = url + "?token" + ip_info_key.as_str(),
+        _ => {}
+      },
+      _ => {}
+    }
```

Comment:
Consider using `if let` over match when there is only one path.  Also consider `format!` over `+` operator
```suggestion
        let mut suffix = String::new();
        if let Some(config) = config {
            if let Some(key) = &config.ip_info_key {
                suffix = format!("?token{}", key);
            }
        }
        let url = format!("https://ipinfo.io/json/{}", suffix);
```